### DISCUSSION TITLE
Appending to pages with metadata

### DISCUSSION THREAD

**Discussion by jrappen** [squidfunk/mkdocs-material#3598] [Q&A] [answered]
I have the following site:

- <https://github.com/da-michele/da-michele.com>
    - <https://github.com/da-michele/da-michele.com/blob/main/docs/src/menu.md>
    - <https://github.com/da-michele/da-michele.com/blob/main/docs/theme/menu.html>
- <https://da-michele.com/menu/>

where I append to a restaurant's menu page the available menu items.

---

I have two questions:

- How do I get search indexing, anchor tracking, etc. working without monkey-patching the page template?
- Is there an easy way to do this with markdown instead of the html template?

**Comment by squidfunk** [comment, marked as answer]
Looks delicious.

1. The search index is created from the Markdown, so  if you're defining stuff via front matter, it can be rendered as part of the page, but will not show up in the search index. Using [Snippets](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/) is probably a better idea.
2. I don't think so.

**Reply by jrappen** [reply]
Thanks for the feedback.

Using **Snippets** works for me.
